server/worker/worker.go

```diff
@@ -0,0 +1,338 @@
+	}
+
+	panelList, ok := panels.([]interface{})
+	if !ok {
+		return emptyResponse
+	}
+
+	return panelList
+}
+
+func isAnalyticsPanel(panel map[string]interface{}) bool {
+	panelType, ok := panel["type"]
+	if ok {
+		return panelType == "macropower-analytics-panel"
+	}
+
+	return false
+}
+
+func (api *Client) getResponseBody(res *http.Response) []byte {
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		level.Info(api.Logger).Log(
+			"status", "error",
+			"message", "Failed to get response body",
+			"error", err,
+		)
+
+		return []byte{}
+	}
+
+	return body
+}
+
+func (api *Client) prepare(method string, endpoint string, payload []byte) (*http.Request, error) {
+	bearer := "Bearer " + api.Token
+
+	req, err := http.NewRequest(method, api.GrafanaUrl+endpoint, bytes.NewBuffer(payload))
+	if err != nil {
+		return nil, errors.New("Failed to create new HTTP Request")
+	}
+
+	req.Header.Add("Authorization", bearer)
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("Accept", "application/json")
+
+	return req, nil
+}
+
+func (api *Client) Get(endpoint string) ([]byte, error) {
+	req, err := api.prepare("GET", endpoint, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{}
+	res, err := client.Do(req)
+	if res.StatusCode != http.StatusOK {
+		return nil, errors.New("Request failed with status: " + strconv.Itoa(res.StatusCode))
+	}
+
+	defer res.Body.Close()
+	if err != nil {
+		return nil, err
+	}
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	return body, nil
+}
+
+func (api *Client) Post(endpoint string, payload []byte) ([]byte, error) {
+	req, err := api.prepare("POST", endpoint, payload)
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{}
+	res, err := client.Do(req)
+	if res.StatusCode != http.StatusOK {
+		return nil, errors.New("Request failed with status: " + strconv.Itoa(res.StatusCode))
+	}
+	defer res.Body.Close()
+
+	if err != nil {
+		return nil, err
+	}
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	return body, nil
+}
+
+func (api *Client) GetDashboard(uid string) *Dashboard {
+	res, err := api.Get("/api/dashboards/uid/" + uid)
+	if err != nil {
+		level.Info(api.Logger).Log(
+			"status", "error",
+			"message", "GetDashboard - api.Get failed",
+			"error", err,
+		)
+
+		return nil
+	}
+
+	var dashboardData map[string]interface{}
+	err = json.Unmarshal(res, &dashboardData)
+	if err != nil {
+		level.Info(api.Logger).Log(
+			"status", "error",
+			"message", "GetDashboard - Failed to parse dashboardData response",
+			"error", err,
+		)
+
+		return nil
+	}
+
+	return &Dashboard{
+		Uid:  uid,
+		Data: dashboardData,
+	}
+}
+
+func (api *Client) updateDashboards(dashboards []Dashboard) {
+	for _, dashboard := range dashboards {
+		if dashboard.Title == "test dashboard - empty" {
+			dashboard.Data["overwrite"] = true
+			dashboard.Data["message"] = "PRO - Auto-add analytics panel"
```

Comment:
this message should be probably configurable, at least we shouldn't include "PRO" in OS code